pyro/primitives.py

```diff
@@ -94,14 +94,14 @@ def _masked_observe(
     name: str,
     fn: TorchDistributionMixin,
     obs: Optional[torch.Tensor],
-    obs_mask: torch.Tensor,
```

Comment:
Nice!  Is `torch.cuda.BoolTensor` a subclass of `torch.BoolTensor`?

Reply:
Yeah, it should be ok. Let's just keep in mind that `assert isinstance(obs_mask, torch.BoolTensor)` would fail in runtime.

Reply:
Yeah. That's why I try to use `assert x is not None` when `x: Optional[torch.Tensor]`